bazel/envoy_library.bzl

```
@@ -101,10 +104,19 @@ def envoy_cc_library(
         strip_include_prefix = None,
         include_prefix = None,
         textual_hdrs = None,
+        alwayslink = None,
         defines = []):
     if tcmalloc_dep:
         deps += tcmalloc_external_deps(repository)
 
+    # If alwayslink is not specified, allow turning it off via --define=library_autolink=disabled
+    # alwayslink is defaulted on for envoy_cc_extensions to ensure the REGISTRY macros work.
+    if not (alwayslink):
```

if this is just the normal `not` keyword I think we need to check against None explicitly, otherwise I think `0` will be false-y

```suggestion
    if alwayslink == None:
```